firebase-firestore/src/test/java/com/google/firebase/firestore/local/CountingQueryEngine.java

```diff
@@ -160,12 +181,19 @@ private DocumentOverlayCache wrapOverlayCache(DocumentOverlayCache subject) {
       @Override
       public Overlay getOverlay(DocumentKey key) {
         ++overlaysReadByKey[0];
-        return subject.getOverlay(key);
+        Overlay overlay = subject.getOverlay(key);
+        overlayTypes.put(key, getOverlayType(overlay));
```

Comment:
for example we'd do: `overlays.put(key, overlay)`

Reply:
See other comment.